Software/lib/GNSE_BSP/GNSE_bsp_clk_timer.c

```diff
@@ -59,6 +70,55 @@ int32_t GNSE_BSP_RTC_Init(void)
   return GNSE_BSP_ERROR_NONE;
 }
 
+/**
+  * @brief  Configures independent watchdog timer and sets sequencer to refresh it
+  * @arg    iwdg_reload: Timeout (in ticks of the IWDG timer) for the IWDG (max is 0xFFF)
+  * @return GNSE_BSP status
+  */
+int32_t GNSE_BSP_IWDG_Init(uint32_t iwdg_reload)
+{
+  /* Set IWDG settings to maximum reload value */
+  GNSE_BSP_iwdg.Instance = IWDG;
+  GNSE_BSP_iwdg.Init.Prescaler = IWDG_PRESCALER_256;
+  GNSE_BSP_iwdg.Init.Window = IWDG_WINR_WIN;
+  GNSE_BSP_iwdg.Init.Reload = iwdg_reload;
+
+  /* Set sequencer on same frequency as IWDG (minus IWDG_SEQ_TRIGGER_MS)
+   * Change calculation when altering the window or prescaler settings
+   */
+  uint32_t IWDG_seq_timeout_ms = (GNSE_BSP_iwdg.Init.Reload / (LSI_VALUE / 256) * 1000);
+  UTIL_TIMER_Create(&IWDG_refresh_timer, 0xFFFFFFFFU, UTIL_TIMER_ONESHOT, GNSE_On_IWDG_Event, NULL);
+  UTIL_TIMER_SetPeriod(&IWDG_refresh_timer, IWDG_seq_timeout_ms);
+  UTIL_TIMER_Start(&IWDG_refresh_timer);
```

Comment:
Please move scheduling to the application layer as the sequencer support is functional on the application layer (bare-metal)

Please adjust the code accordingly.